### ISSUE TITLE
Improve filtering annotation

### ISSUE THREAD

**Issue by alexander-yevsyukov** [SpineEventEngine/core-java#1145]
In order to filter subscribed events by a field value, we use the `filter` field of the `@Subscribe` annotation. It looks like this:

```java
@Subscribe(filter = @ByField(
        path = "handled_signal.id.type_url",
        value = "type.spine.io/spine.core.Command"
))
void on(HandlerFailedUnexpectedly event) {
...
} 
```

This works OK, but there are issues associated with this declaration:
  1. The annotation value (which is also an annotation) stays too far from the parameter (`event)` it is related to.
  2. It's unclear if the filter accepts or rejects the values.
  3. The declaration is a bit lengthy.

We can address this by introducing a parameter annotation, which may look like this:

```java
@Subscribe
void on(@Where(field = "handled_signal.id.type_url",
                equals = "type.spine.io/spine.core.Command")
        HandlerFailedUnexpectedly event) {
    ...
}
```

**Comment by alexander-yevsyukov** [comment]
@armiol, @dmdashenkov, have a look at this improvement request from me and let's discuss.

**Comment by alexander-yevsyukov** [comment]
The referenced code is  from `io.spine.system.server.CommandLogProjection`.

**Comment by alexander-yevsyukov** [comment]
Addressed by #1159.
